File: benchmarks/julia/b5_regex.jl
# Hot loop in a typed function so eachline/occursin run with concrete types
# rather than dispatching through Any-typed globals.
function count_matches(io::IO, pat::Regex)
    count = 0
    for line in eachline(io)
        if occursin(pat, line)
            count += 1
        end
    end
    return count
end

function main()
    fn = length(ARGS) > 0 ? ARGS[1] : "../../data/regex_input.txt"
    pat = r"[a-zA-Z0-9._%+\-]+@[a-zA-Z0-9.\-]+\.[a-zA-Z]{2,}"
    n = open(io -> count_matches(io, pat), fn)
    println(n)
end

main()

File: benchmarks/python/b5_regex.py
import re, sys
fn = sys.argv[1] if len(sys.argv) > 1 else "../../data/regex_input.txt"
pat = re.compile(r"[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}")
c = 0
with open(fn) as f:
    for line in f:
        if pat.search(line): c += 1
print(c)
